lib/stdlib/src/peer.erl

```diff
@@ -0,0 +1,909 @@
+verify_args(Options) ->
+    %% verify that "Args" is valid - common problem is when Args aren't strings
+    Args = maps:get(args, Options, []),
+    is_list(Args) orelse error({invalid_arg, Args}),
+    [error({invalid_arg, Arg}) || Arg <- Args, not io_lib:char_list(Arg)],
+    %% alternative connection must be requested for non-distributed node,
+    %%  or a distributed node when origin is not alive
+    is_map_key(connection, Options) orelse
+        (is_map_key(name, Options) andalso erlang:is_alive()) orelse error(not_alive),
+    %% exec must be a string, or a tuple of string(), [string()]
+    case maps:find(exec, Options) of
+        {ok, {Exec, Strs}} ->
+            io_lib:char_list(Exec) orelse error({exec, Exec}),
+            [error({exec, Str}) || Str <- Strs, not io_lib:char_list(Str)],
+            ok;
+        {ok, Exec} when is_list(Exec) ->
+            io_lib:char_list(Exec) orelse error({exec, Exec}),
+            ok;
+        error ->
+            ok;
+        {ok, Err} ->
+            error({exec, Err})
+    end.
+
+make_notify_ref(infinity) ->
+    {self(), make_ref()};
+make_notify_ref(WaitBoot) when is_integer(WaitBoot) ->
+    {self(), make_ref()};
+make_notify_ref({ReplyTo, Tag}) when is_pid(ReplyTo) ->
+    {ReplyTo, Tag};
+make_notify_ref(false) ->
+    false.
+
+start_it(Options, StartFun) ->
+    verify_args(Options),
+    WaitBoot = maps:get(wait_boot, Options, ?WAIT_BOOT_TIMEOUT),
+    Notify = make_notify_ref(WaitBoot),
+    case gen_server:StartFun(?MODULE, [Notify, Options], []) of
+        {ok, Pid} when WaitBoot =:= infinity ->
+            {_, Ref} = Notify,
+            Mref = erlang:monitor(process, Pid),
+            receive
+                {Ref, {started, NodeName, Pid}} ->
+                    erlang:demonitor(Mref, [flush]),
+                    {ok, Pid, NodeName};
+                {Ref, {boot_failed, Reason, Pid}} ->
+                    erlang:demonitor(Mref, [flush]),
+                    erlang:exit({boot_failed, Reason});
+                {'DOWN', Mref, _, _, Reason} ->
+                    erlang:exit(Reason)
+            end;
+        {ok, Pid} when is_integer(WaitBoot) ->
+            {_, Ref} = Notify,
+            Mref = erlang:monitor(process, Pid, [{alias, demonitor}]),
+            receive
+                {Ref, {started, NodeName, Pid}} ->
+                    erlang:demonitor(Mref, [flush]),
+                    {ok, Pid, NodeName};
+                {Ref, {boot_failed, Reason, Pid}} ->
+                    erlang:demonitor(Mref, [flush]),
+                    erlang:exit({boot_failed, Reason});
+                {'DOWN', Mref, _, _, Reason} ->
+                    erlang:exit(Reason)
+            after
+                WaitBoot ->
+                    _ = gen_server:stop(Pid),
+                    erlang:demonitor(Mref, [flush]),
+                    receive
+                        {[alias | Mref], Reply} ->
```

Comment:
This `receive` clause should never be able to match anything since `Mref` isn't communicated with any other processes (unless `Mref` should be picked up via `process_info()` which would be a bit strange). Since `Mref` is not communicated to anyone, there is also no reason to make it an alias since noone will be able to use it as an alias.